nucliadb/nucliadb/ingest/service/writer.py

```diff
@@ -759,21 +763,15 @@ async def ReIndex(self, request: IndexResource, context=None) -> IndexStatus:  #
                 await kbobj.set_resource_shard_id(request.rid, shard.sharduuid)
 
             if shard is not None:
-                counter = await shard.add_resource(
+                await shard.add_resource(
                     brain.brain,
                     0,
                     partition=self.partitions[0],
                     kb=request.kbid,
                     reindex_id=uuid.uuid4().hex,
                 )
-
-                if counter is not None and counter.fields > settings.max_shard_fields:
-                    # check to see if we've exceeded the max resources per shard
-                    # and create a new shard for new resources to land on
-                    similarity = await kbobj.get_similarity()
-                    shard = await node_klass.create_shard_by_kbid(
-                        txn, request.kbid, similarity=similarity
-                    )
+                # trigger possibly creating new shard manually here
+                await self.sc.process_kb(request.kbid)
```

Comment:
I'm not sure we should import and use this here.

Indexing will already be listening to indexed events and be doing this new shard logic.

In fact, it could cause problems if we do it in 2 places right?

Reply:
🤦🏼 you're right! I forgot that the notifications are sent via nats pubsub. removed!